src/pages/products-listing/products-listing.jsx

```diff
@@ -0,0 +1,94 @@
+import './products-listing.css';
+//import data from '../../data';
+import { Card } from '../../components/Card/Card';
+import { Filter } from '../../components/Filters/Filter'
+
+import { useCart } from '../../context/cart-context';
+import { useProduct } from '../../context/product-context';
+import { Navbar } from '../../components/Navbar/navbar'
+
+export function ProductListing() {
+
+    const { cartList, wishList, dispatch } = useCart();
+    const { productList, sortBy, showAllInventory,
+        showFastDelivery, searchTerm, productFilterDispatch } = useProduct();
+
+    function getColor(wishList, id) {
+        const index = wishList.findIndex((wishlistItem) => wishlistItem._id === id);
+        if (index !== -1) {
+            return "#F87171";
+        }
+        return "#E5E7EB";
+    }
+    function isItemInWishList(wishList, id) {
+
+        const index = wishList.findIndex((wishlistItem) => wishlistItem._id === id);
+        if (index !== -1) {
+            return true;
+        }
+        return false;
+    }
+    function isItemInCart(cartList, id) {
+        const index = cartList.findIndex((cartListItem) => cartListItem._id === id);
+        if (index !== -1) {
+            return true;
+        }
+        return false;
+    }
+
```

Comment:
These two functions can be made into one:
keep this in a separate file and reuse it anywhere u like !! 
```const checkIfAlreadyPresent = (array, itemId) => {
  const isPresent = array.find((product) => product.id  === itemId);
  return isPresent;
};```
I have suggested `find` method here.